src/librustc_mir/interpret/operand.rs

```diff
@@ -190,6 +187,19 @@ impl<'tcx, Tag: Copy> ImmTy<'tcx, Tag>
     }
 }
 
+impl<'tcx, Tag> OpTy<'tcx, Tag>
+{
+    /// This is used by [priroda](https://github.com/oli-obk/priroda) to create an `OpTy`.
+    ///
+    /// Do not use this from inside rustc.
```

Comment:
Can you expand a bit on why this shouldn't be used from inside rustc? Is it just that this operation is unchecked or does it violate some other assumption?

Reply:
I thought that somebody wanted to prevent having a method like this to prevent accidental misusage or something like that, but in https://github.com/rust-lang/rust/pull/55179 I can really find anything like that except for "I wanted to have something like an abstraction here" by @RalfJung.

Reply:
The `*Ty` types have successfully found and prevent several bugs. Currently the only "hole" in the scheme is `ImmTy`, you can construct arbitrary immediates from the outside. I was very happy when I managed to make `OpTy` private and only expose `ImmTy` instead, having to un-do this would be very sad. :(

Reply:
Really the advice here should be "do not use this, period". This is not an inside-rustc vs outside-rustc distinction.